publisher/src/main/java/com/pavlovmedia/oss/jaxrs/publisher/impl/JerseyPublisher.java

```diff
@@ -188,16 +187,14 @@ protected void activate(final Map<String,Object> properties, final BundleContext
      * is for SSE, it is listed as an optional import in the manifest
      * so one may choose not to bring it in, and this should allow the
      * system to run, even without it.
-     * 
-     * @param featureClassSupplier A lambda that returns a class to register
      */
     // This catches Exception because OSGi can make this fail in unusual ways
     private void tryRegisterFeature(final Supplier<Class<?>> featureClassSupplier) {
         try {
-            Object feature = featureClassSupplier.get().newInstance();
-            ServiceRegistration reg = bundleContext.registerService(featureClassSupplier.get().getName(), feature, null);
+            Object feature = featureClassSupplier.get().getDeclaredConstructor().newInstance();
+            ServiceRegistration<?> reg = bundleContext.registerService(featureClassSupplier.get().getName(), feature, null);
             featureRegistrations.add(reg);
-        } catch (NoClassDefFoundError | InstantiationException | IllegalAccessException e) {
+        } catch (Exception e) {
```

Comment:
We shouldn't be catching the generic Exception without a good reason, is there a good reason here?

Reply:
Java 11 does not support 'featureClassSupplier.get().newInstance()', so I changed it to 'featureClassSupplier.get().getDeclaredConstructor().newInstance()'. This made me catch 4 more exceptions (InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException). The existing try catch was catching 3 exceptions, so instead of catching 7 different exceptions, I used the generic Exception. However, this doesn't seem to be a good reason, so I will change it to catch specific exceptions instead of generic.